match.py

```diff
@@ -0,0 +1,176 @@
+            return False
+    if not wildcards & FlowWildCards.OFPFW_DL_VLAN:
+        if 'vlan_vid' not in args:
+            return False
+        if match_fields.get('vlan_vid') != args.get('vlan_vid')[-1]:
+            return False
+    if not wildcards & FlowWildCards.OFPFW_DL_SRC:
+        if 'eth_src' not in args:
+            return False
+        if match_fields.get('eth_src') != args.get('eth_src'):
+            return False
+    if not wildcards & FlowWildCards.OFPFW_DL_DST:
+        if 'eth_dst' not in args:
+            return False
+        if match_fields.get('eth_dst') != args.get('eth_dst'):
+            return False
+    if not wildcards & FlowWildCards.OFPFW_DL_TYPE:
+        if 'eth_type' not in args:
+            return False
+        if match_fields.get('eth_type') != int(args.get('eth_type')):
+            return False
+    if not _match_ipv4_10(match_fields, args, wildcards):
+        return False
+    return flow_dict
+
+
+def match13_no_strict(flow_to_install, stored_flow_dict):
+    """Match a packet againts the stored flow (OF 1.3).
+
+    Return the flow if any fields match, otherwise, return False.
+    """
+    if flow_to_install.get('cookie_mask') and 'cookie' in stored_flow_dict:
+        cookie = flow_to_install['cookie'] & flow_to_install['cookie_mask']
+        stored_cookie = (stored_flow_dict['cookie'] &
+                         flow_to_install['cookie_mask'])
+        if cookie == stored_cookie:
+            return stored_flow_dict
+        return False
+    if 'match' not in flow_to_install:
+        return False
+
+    for key, value in flow_to_install.get('match').items():
+        if 'match' not in stored_flow_dict:
+            return False
+        if key not in ('ipv4_src', 'ipv4_dst', 'ipv6_src', 'ipv6_dst'):
+            if value == stored_flow_dict['match'].get(key):
+                return stored_flow_dict
+        else:
+            field = stored_flow_dict['match'].get(key)
+            if not field:
+                return False
+            masked_ip_addr = ipaddress.ip_network(value, False)
+            field_mask = field + "/" + str(masked_ip_addr.netmask)
+            masked_stored_ip = ipaddress.ip_network(field_mask, False)
+            if masked_ip_addr == masked_stored_ip:
+                return stored_flow_dict
+    return False
+
+
+def match_flows(flow_to_install, version, stored_flow_dict):
+    # pylint: disable=bad-staticmethod-argument
+    """
+    Match the packet in request against the flows installed in the switch.
+
+    Try the match with each flow, in other. If many is True, tries the
+    match with all flows, if False, tries until the first match.
+    :param args: packet data
+    :param many: Boolean, indicating whether to continue after matching the
+            first flow or not
+    :return: If many, the list of matched flows, or the matched flow
+    """
+    match = do_match(flow_to_install, version, stored_flow_dict)
+    return match
```

Comment:
1. All this functions does is call `do_match`
1. The docstring for it is completely wrong.

My vote is to rename `do_match` to `match_flow` (singular) and use it in the other modules.

Reply:
Resolved in https://github.com/kytos/flow_manager/pull/117/commits/4b1644d53e7617efd6ba030ffbdbc33671156e86